### ISSUE TITLE
Write function that collects historical prices of stocks and crypto

### ISSUE THREAD

**Issue by tyrovirtuosso** [tyrovirtuosso/MarketSurfer#1]
Use APIs the fetch crypto and stock data.

Crypto:
Coingecko, CCXT

Stocks:
yfinance, Alpaca

**Comment by tyrovirtuosso** [comment]
Decided to try FastAPI instead of django or flask, since its modern, light weight, and allows asynchronous functions

**Comment by tyrovirtuosso** [comment]
I got a bit carried away with fastapi and websockets plus i got fever. Seems its taking too much of my time trying to write scripts to take in the inputs. Instead, I should solely be focused on fetching data and storing it in a database or hdfs system

**Comment by tyrovirtuosso** [comment]
**Data Collection:**
Implemented a data collector that collects stock data from Alpaca API and crypto data from Coingecko. 1hr interval data is collected for both. I have added the modularity and flexibility to support other data sources like yfinance, ccxt and more.

Coingecko API has both a limit on the amount of data per call and no. of API calls. Requested max data per cal
and used VPN to bypass the API call limit.

**Data Processing:**
Normalized the data and synchronized the columns among stock and crypto. Crypto is 24/7, while stocks have holidays. 
Used Linear Interpolation to resample the missing stock data to create uniformity among stock and crypto.

**Data Storage**:
 Data can currently be stored in an SQLite database or as a CSV in the file system. I have made it modular and flexible to add more collection methods. I had initiated the HDFS storage method, but the pyarrow package used to connect Python with HDFS is deprecated and only runs on a Java version not supported by Mac. I have decided not worth the effort to use HDFS.
